### PR TITLE
Example demonstrating S3 and Cached File providers

### PR DESCRIPTION
**Pull request by adam-fowler**

### CODE REVIEW

File: s3-file-provider/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# ================================
+# Build image
+# ================================
+FROM swift:5.10-jammy as build
```
Comment (Joannis) [inline s3-file-provider/Dockerfile:4]: Shouldn't (new) examples become swift-6 now?

File: s3-file-provider/Package.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
+// swift-tools-version:5.9
```
Comment (Joannis) [inline s3-file-provider/Package.swift:1]: Likewise, shouldn't we prioritise Swift 6 now?

File: s3-file-provider/Sources/App/Application+build.swift [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+///  Build application
+/// - Parameter arguments: application arguments
+public func buildApplication(_ arguments: some AppArguments, environment: Environment) async throws -> some ApplicationProtocol {
+    let logger = {
+        var logger = Logger(label: "s3_file_provider")
+        logger.logLevel = 
+            arguments.logLevel ??
+            environment.get("LOG_LEVEL").map { Logger.Level(rawValue: $0) ?? .info } ??
+            .info
```
Comment (Joannis) [inline s3-file-provider/Sources/App/Application+build.swift:27]:
```suggestion
            environment.get("LOG_LEVEL").flatMap(Logger.Level.init) ?? .info
```

**Review by Joannis — APPROVED** [review]
Excellent. Just a note on the Swift version being used

File: s3-file-provider/Dockerfile (hunk 1 above)
Comment (adam-fowler) [reply s3-file-provider/Dockerfile:4]: Yep that came from the template which I'm just going to update

File: s3-file-provider/Package.swift (hunk 2 above)
Comment (adam-fowler) [reply s3-file-provider/Package.swift:1]: yep

**Pull request merged**